### ISSUE TITLE
[Bug]: attach_data_node updates dimension partitions with repartition=>false

### ISSUE THREAD

**Issue by matthewmcnew** [timescale/timescaledb#5157]
### What type of bug is this?

Incorrect result

### What subsystems and features are affected?

Partitioning

### What happened?

`attach_data_node` updates the `dimension_partition` with `repartition => false`. 

### TimescaleDB version affected

2.9.0

### PostgreSQL version used

14.6

### What operating system did you use?

Mac OS X

### What installation method did you use?

Docker

### What platform did you run on?

On prem/Self-hosted

### How can we reproduce the bug?

```bash
1. Add a data nodes:

select add_data_node('dn-1', host => 'dn-1');
```
2. Create a Distributed Hypertable:
```
CREATE TABLE test
(
    time   timestamptz not null,
    device int
);

SELECT
FROM public.create_distributed_hypertable(
        'test',
        'time',
        number_partitions => 4,
        partitioning_column => 'device'
    );
```
3. View the dimension partitions:
```
SELECT * FROM _timescaledb_catalog.dimension_partition;
```
| dimension\_id | range\_start | data\_nodes |
| :--- | :--- | :--- |
| 4 | -9223372036854775808 | {dn-1} |
| 4 | 536870911 | {dn-1} |
| 4 | 1073741822 | {dn-1} |
| 4 | 1610612733 | {dn-1} |

4. Add another data node
```
select add_data_node('dn-2', host => 'dn-2');
```
5. Attach the data node to the hypertable with repartition false:
```
select attach_data_node('dn-2', 'test',  repartition => false);
```
6. View the updated and dimension partitions:
```
SELECT * FROM _timescaledb_catalog.dimension_partition;
```
| dimension\_id | range\_start | data\_nodes |
| :--- | :--- | :--- |
| 2 | -9223372036854775808 | {dn-1} |
| 2 | 1073741823 | {dn-2} |
```

**Comment by fabriziomello** [comment]
@pmwkaa can u have a look on this one?

**Comment by erimatnor** [comment]
@matthewmcnew should be fixed by #5193 

Let us know if that's not the case.
